cdap-ui/app/cdap/components/CaskWizards/AddNamespace/index.js

```diff
@@ -13,32 +13,71 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
-import PropTypes from 'prop-types';
 
+import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import WizardModal from 'components/WizardModal';
 import Wizard from 'components/Wizard';
 import AddNamespaceStore from 'services/WizardStores/AddNamespace/AddNamespaceStore';
 import AddNamespaceActions from 'services/WizardStores/AddNamespace/AddNamespaceActions';
 import AddNamespaceWizardConfig from 'services/WizardConfigs/AddNamespaceWizardConfig';
 import NamespaceStore from 'services/NamespaceStore';
-import { PublishNamespace, PublishPreferences } from 'services/WizardStores/AddNamespace/ActionCreator';
+import {PublishNamespace, PublishPreferences, editNamespaceProperties} from 'services/WizardStores/AddNamespace/ActionCreator';
 import T from 'i18n-react';
 import {Observable} from 'rxjs/Observable';
+import isEmpty from 'lodash/isEmpty';
 
 export default class AddNamespaceWizard extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showWizard: this.props.isOpen,
-      successInfo: {}
-    };
+  state = {
+    showWizard: this.props.isOpen,
+    successInfo: {}
+  };
+
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    context: PropTypes.string,
+    onClose: PropTypes.func,
+    isEdit: PropTypes.bool,
+    editableFields: PropTypes.array,
+    properties: PropTypes.object,
+    activeStepId: PropTypes.string
+  };
+
+  static defaultProps = {
+    isEdit: false,
+    editableFields: [],
+    properties: {}
+  };
+
+  componentWillMount() {
+    if (this.props.properties && !isEmpty(this.props.properties)) {
+      AddNamespaceStore.dispatch({
+        payload: { ...this.props.properties },
+        type: AddNamespaceActions.setProperties
+      });
+    }
+    if (this.props.editableFields && this.props.editableFields.length) {
+      AddNamespaceStore.dispatch({
+        payload: { editableFields: this.props.editableFields },
+        type: AddNamespaceActions.setEditableFields
+      });
+    }
   }
+
   componentWillReceiveProps({isOpen}) {
     this.setState({
       showWizard: isOpen
     });
   }
+
+  onSubmit = () => {
+    if (this.props.isEdit) {
+      return editNamespaceProperties();
+    } else {
+      return this.createNamespace();
```

Comment:
should this use `PublishNamespace` from the namespace action creator?

Reply:
Internally `this.createNamespace()` does use `PublishNamespace()`. Have renamed `this.createNamespace()` to `this.publishNamespaceAndPreferences()` to be more clear.